diff --git a/DMXKeyboard.cpp b/DMXKeyboard.cpp
index cccad31..306b299 100644
--- a/DMXKeyboard.cpp
+++ b/DMXKeyboard.cpp
@@ -57,13 +57,13 @@ int DMXKeyboard::KeySymToScanCode(rfbKeySym key)
        }
        else if (code >= 0xFFE1 && code <= 0xFFEE) {
                static const uint16_t map[] =
-               { KEY_LEFTSHIFT, KEY_LEFTSHIFT,
-                       KEY_LEFTCTRL, KEY_LEFTCTRL,
-                       KEY_LEFTSHIFT, KEY_LEFTSHIFT,
+               { KEY_LEFTSHIFT, KEY_RIGHTSHIFT,
+                       KEY_LEFTCTRL, KEY_RIGHTCTRL,
+                       KEY_CAPSLOCK, KEY_LEFTSHIFT,
                        0, 0,
                        KEY_LEFTALT, KEY_RIGHTALT,
                        0, 0, 0, 0 };
-               scancode = map[code & 0xF];
+               scancode = map[(code & 0xF) - 1];
        }
        else if ((code >= 'A' && code <= 'Z') || (code >= 'a' && code <= 'z')) {
                static const uint16_t map[] = {
